core/ulog/log.go

```diff
@@ -60,21 +60,45 @@ type baselogger struct {
 type Log interface {
 	// Configure Log object with the provided log.Configuration
 	Configure(Configuration)
-	// Create a child logger with some context
-	With(...interface{}) Log
-	SetLevel(zap.Level)
-	SetLogger(zap.Logger)
-	Check(zap.Level, string) *zap.CheckedMessage
+
+	// With creates a child logger with the provided parameters as key value pairs
```

Comment:
please explain what that means semantically. Does the child logger appends these k-v pairs to any other k-v pairs passed to logging methods? Or does it do something different with them?